react/components/RenderProvider.tsx

```diff
@@ -374,6 +403,13 @@ class RenderProvider extends Component<Props, RenderProviderState> {
     })
   }
 
+  public addMessages = (newMessages: Record<string, string>) => {
```

Comment:
Instead of having to `addMessages`, I'd try to merge `getCustomMessages` into the `messages` passed to `__provideRuntime`.